client/context/context.go

```diff
@@ -254,3 +256,28 @@ func (ctx CLIContext) WithSimulation(simulate bool) CLIContext {
 	ctx.Simulate = simulate
 	return ctx
 }
+
+// PrintOutput prints output while respecting output and indent flags
+// NOTE: pass in marshalled structs that have been unmarshaled
+// because this function will panic on marshaling errors
+func (ctx CLIContext) PrintOutput(toPrint fmt.Stringer) (err error) {
```

Comment:
If you are going to review this PR, understanding this function is the way to start. The rest of the changes just implement this function at the end of each `gaiacli query` command.